### PR TITLE
Camera model

### PR DESCRIPTION
**Pull request by AugustDailey**

### CODE REVIEW

**Review by lyonsmj — CHANGES_REQUESTED** [review]
z axis bad
add camera behavior, move gamestate update there
use level sizes in enemy proc gen, not camera size

File: FuncGameDev/FSharp/Behavior/CameraBehavior.fs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+﻿module CameraBehavior
+
+let updateCamera (pos : float*float) (rot : float*float) (scale : float*float) (width : float) (height : float) =
```
Comment (lyonsmj) [inline FuncGameDev/FSharp/Behavior/CameraBehavior.fs]: You pass in a width and height but never use them? Also, it's not a valid command function unless its last argument is a gamestate.

File: FuncGameDev/FSharp/Behavior/CameraBehavior.fs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    let cScaleY = snd scale
+
+
+    let cd : CameraData.T = { 
+        position = (cPosX, cPosY);
+        rotation = (cRotX, cRotY);
+        scale = (cScaleX, cScaleY);
+        width = 20.0;
+        height = 10.0 }
+    let gd : GameData.T = {
```
Comment (lyonsmj) [inline FuncGameDev/FSharp/Behavior/CameraBehavior.fs]: This is overkill. Just use the `with` keyword. Something like 'gd = { gd with camera = cd }', I think. Should probably do that for cd, too, but since we're setting every field right now it doesn't really matter.

File: FuncGameDev/FSharp/Behavior/CameraBehavior.fs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        time = GameState.instance.gamedata.time;
+        floor = GameState.instance.gamedata.floor;
+        camera = cd }
+
+    let gs = GameState.instance
+    GameState.instance <- { gs with gamedata = gd}
```
Comment (lyonsmj) [inline FuncGameDev/FSharp/Behavior/CameraBehavior.fs]: This is not what a command function is supposed to do. A command function should return a gamestate, not set the gamestate itself.

File: FuncGameDev/FSharp/Unity/Updater/CameraUpdater.fs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+    let cRotX = transform.rotation.x |> float
+    let cRotY = transform.rotation.y |> float
+    let rot = (cRotX, cRotY)
+
+    let cScaleX = transform.localScale.x |> float
+    let cScaleY = transform.localScale.y |> float
+    let scale = (cScaleX, cScaleY)
+
+    CameraBehavior.updateCamera pos, rot, scale, 20.0, 10.0
```
Comment (lyonsmj) [inline FuncGameDev/FSharp/Unity/Updater/CameraUpdater.fs]: Why are you hardcoding width and height to 20? Also, this is not the right place to call the CameraBehavior, though frankly the organization of our gamestate update calls isn't great to begin with. Still: these updaters are for Unity GameObjects, not for the gamestate. These should be separate.

File: FuncGameDev/FSharp/Unity/Update.fs [hunk 5]
```diff
@@ -15,6 +15,7 @@ type Updater() =
         ()
 
     member this.Update() =
+        CameraUpdater.update
```
Comment (lyonsmj) [inline FuncGameDev/FSharp/Unity/Update.fs]: Move this later. We should update the gamestate before we update any GameObjects. Actually, frankly we should separate them out, but I won't force you to do that here-- just move this call next to where we update the other gameobjects.

Comment (AugustDailey) [reply FuncGameDev/FSharp/Unity/Update.fs]: Oops, my b called the wrong func

File: FuncGameDev/FSharp/Unity/Updater/CameraUpdater.fs [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+    let cRotX = transform.rotation.x |> float
+    let cRotY = transform.rotation.y |> float
+    let rot = (cRotX, cRotY)
+
+    let cScaleX = transform.localScale.x |> float
+    let cScaleY = transform.localScale.y |> float
+    let scale = (cScaleX, cScaleY)
+
+    CameraBehavior.updateCamera pos rot scale 20.0 10.0 |> Commands.addCommand
```
Comment (lyonsmj) [inline FuncGameDev/FSharp/Unity/Updater/CameraUpdater.fs]: We should add this command where we add the other commands in the update loop. Right now it seems like you're trying to update the gamestate based on what's in the gameobject, which is opposite our current design.

File: FuncGameDev/FSharp/Unity/CameraController.fs [hunk 7]
```diff
@@ -0,0 +1,25 @@
+﻿module CameraController
```
Comment (lyonsmj) [inline FuncGameDev/FSharp/Unity/CameraController.fs]: What does this file do/where is it used?

File: FuncGameDev/FSharp/Unity/Update.fs [hunk 8]
```diff
@@ -15,6 +15,7 @@ type Updater() =
         ()
 
     member this.Update() =
+        CameraInterpreter.updateCamera
```
Comment (lyonsmj) [inline FuncGameDev/FSharp/Unity/Update.fs]: Why is this here? This is literally just a function, right? You're not even calling it, just stating it?

**Review by lyonsmj — APPROVED** [review]
